servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcMessageEncodings.java

```diff
@@ -15,72 +15,81 @@
+import io.servicetalk.http.api.DefaultStreamingContentCodecBuilder.DeflateStreamingContentCodecBuilder;
+import io.servicetalk.http.api.DefaultStreamingContentCodecBuilder.GzipStreamingContentCodecBuilder;
+import io.servicetalk.http.api.IdentityContentCodec;
+
 import java.util.Collection;
-import java.util.HashSet;
-import java.util.Set;
 import javax.annotation.Nullable;
 
-import static java.util.Arrays.asList;
-import static java.util.Collections.unmodifiableSet;
+import static io.servicetalk.http.api.CharSequences.contentEqualsIgnoreCase;
+import static io.servicetalk.http.api.CharSequences.isEmpty;
+import static io.servicetalk.http.api.CharSequences.startsWith;
 import static java.util.Objects.requireNonNull;
 
 /**
  * Default available encoding implementations.
  * Encoding {@link #identity()} is always supported regardless of the client or server settings.
- *
- * {@link #all()} is a set that includes all default encodings {@link #deflate()} and {@link #gzip()}.
  */
 public final class GrpcMessageEncodings {
 
-    private static final GrpcMessageEncoding IDENTITY =
-            new DefaultGrpcMessageEncoding("identity", new IdentityMessageCodec());
-
-    private static final GrpcMessageEncoding GZIP =
-            new DefaultGrpcMessageEncoding("gzip", new GzipMessageCodec());
+    private static final ContentCodec IDENTITY = new IdentityContentCodec();
 
-    private static final GrpcMessageEncoding DEFLATE =
-            new DefaultGrpcMessageEncoding("deflate", new DeflateMessageCodec());
+    private static final ContentCodec GZIP = new GzipStreamingContentCodecBuilder().build();
 
-    private static final Set<GrpcMessageEncoding> ALL =
-            unmodifiableSet(new HashSet<>(asList(IDENTITY, GZIP, DEFLATE)));
+    private static final ContentCodec DEFLATE = new DeflateStreamingContentCodecBuilder().build();
 
     private GrpcMessageEncodings() {
     }
 
     /**
-     * Returns the default, always supported 'identity' {@link GrpcMessageEncoding}.
-     * @return the default, always supported 'identity' {@link GrpcMessageEncoding}
+     * Returns the default, always supported 'identity' {@link ContentCodec}.
+     * @return the default, always supported 'identity' {@link ContentCodec}
      */
-    public static GrpcMessageEncoding identity() {
+    public static ContentCodec identity() {
         return IDENTITY;
     }
 
     /**
-     * Returns a GZIP based {@link GrpcMessageEncoding} backed by {@link java.util.zip.Inflater}.
-     * @return a GZIP based {@link GrpcMessageEncoding} backed by {@link java.util.zip.Inflater}
+     * Returns a GZIP based {@link ContentCodec} backed by {@link java.util.zip.Inflater}.
+     * The max allowed payload size for this codec is 2Mib.
+     *
+     * @return a GZIP based {@link ContentCodec} backed by {@link java.util.zip.Inflater}
      */
-    public static GrpcMessageEncoding gzip() {
+    public static ContentCodec gzipDefault() {
```

Comment:
Can users use a factory from `http-api`?

Reply:
Yes we can, especially if we merged the codec APIs then its even better.